src/inspect.rs

```diff
@@ -1,24 +1,202 @@
 use anyhow::Result;
 use biscuit_auth::{
-    builder::{Fact, Policy, Rule},
+    builder::{Fact, Rule},
     datalog::RunLimits,
     error::{FailedCheck, Logic, MatchedPolicy, RunLimit, Token},
     Authorizer, UnverifiedBiscuit,
 };
 use chrono::offset::Utc;
+use serde::Serialize;
 use std::fs;
 use std::path::PathBuf;
 
 use crate::cli::*;
 use crate::errors::CliError::*;
 use crate::input::*;
 
+#[derive(Serialize)]
+struct TokenBlock {
+    code: String,
+    external_key: Option<String>,
+    revocation_id: String,
+}
+
+#[derive(Serialize)]
+struct TokenDescription {
+    sealed: bool,
+    root_key_id: Option<u32>,
+    blocks: Vec<TokenBlock>,
+}
+
+impl TokenDescription {
+    fn render(&self) {
+        if self.sealed {
+            println!("Sealed biscuit");
+        } else {
+            println!("Open biscuit");
+        }
+
+        for (i, block) in self.blocks.iter().enumerate() {
```

Comment:
Rather than chaining `println!`(which triggers an IO every time), shouldn't be interesting to build the full text with a string_builder for example ?

Reply:
agreed, i had planned it for a follow-up PR as it will require updating existing code that is not touched by this PR yet.

Reply:
👌👌